File: symmerticTree/symTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        if root == None:return True
        else: return self.isSubSymmetric(root.left,root.right)
    def isSubSymmetric(self,left:TreeNode,right:TreeNode)->bool:
        if left == right == None:return True
        if left == None or right == None:return False
        if left.val == right.val:
            return self.isSubSymmetric(left.left,right.right) and self.isSubSymmetric(left.right,right.left)
        return False

File: symmerticTree/symTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root == null) return true;
        else return isSubSymmetric(root.left,root.right);
    }
    public boolean isSubSymmetric(TreeNode left,TreeNode right){
        if(left==null&&right==null) return true;
        if(left==null||right==null) return false;
        if(left.val ==right.val)
            return isSubSymmetric(left.left,right.right)&&isSubSymmetric(left.right,right.left);
        return false;
    }
}